vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/order_handler.rb

```diff
@@ -17,29 +17,33 @@ def initialize(occupancy)
       def process
         return unless occupancy.account_id?
 
-        ActiveRecord::Base.transaction do
-          create_order
-          create_order_detail
-
-          order.validate_order!
-          order.purchase!
-        end
+        create_order unless occupancy.order_detail_id?
```

Comment:
I think this makes sense, I think these methods are just a little awkward. What if this were called `find_or_create_order` and push the predicate into that method?